### DISCUSSION TITLE
post/ipv6-tcp-hole-punching/

### DISCUSSION THREAD

**Discussion by giscus** [lyc8503/lyc8503#14] [Announcements]
# post/ipv6-tcp-hole-punching/

某天我在使用 syncthing 同步时, 发现外网的主机 (称为B) 成功连接到了我家防火墙内的主机 (称为 A). 当时我检查了主机 A 所处网络的 OpenWRT 路由器的防火墙设置. 设置为 Input reject, Output accept, Forward reject. 理论上路由器不会允许外网主机 B 直接访问主机 A, 可它却连上了… 众所周知, 困扰程序员的两大问题就是它为

https://blog.lyc8503.site/post/ipv6-tcp-hole-punching/

**Comment by HHHHao-s** [comment]
cool
